main.tf

```diff
@@ -0,0 +1,200 @@
+// Define Local Vars
+
+locals {
+  tags = {
+    Terraform   = "${var.terraform_tag}"
+    Environment = "${var.environment}"
+  }
+}
+
+data "aws_region" "current" {}
+
+// VPC Configuration
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~>3.18.1"
+
+  name = "${var.prefix}-spot-instance-vpc-${var.environment}"
+  cidr = var.use_ipam_pool ? null : var.vpc_cidr_block
+
+  azs            = ["${data.aws_region.current.name}a", ]
+  public_subnets = [var.public_subnet_cidr_block, ]
+
+  tags = local.tags
+}
+
+// Instance Configuration
+
+resource "aws_key_pair" "instance_ssh_key" {
+  key_name   = "${module.vpc.name}-ssh-key-pair"
```

Comment:
should probably use prefix

Reply:
the prefix is included in the module.vpcname

Reply:
You just mixed up the naming convention which was {prefix}-NAME-{environment}

Reply:
naming convention applied to resource name